bundles/org.openhab.binding.revogi/src/main/java/org/openhab/binding/revogi/internal/udp/UdpSenderService.java

```diff
@@ -0,0 +1,135 @@
+ */
+@NonNullByDefault
+public class UdpSenderService {
+
+    /**
+     * Limit timeout waiting time, as we have to deal with UDP
+     *
+     * How it works: for every loop, we'll wait a bit longer, so the timeout counter is multiplied with the timeout base
+     * value. Let max timeout count be 2 and timeout base value 800, then we'll have a maximum of loops of 3, waiting
+     * 800ms in the 1st loop, 1600ms in the 2nd loop and 2400ms in the third loop.
+     */
+    private static final int MAX_TIMEOUT_COUNT = 2;
+    public static final long TIMEOUT_BASE_VALUE = 800L;
+    private static final int REVOGI_PORT = 8888;
+
+    private final Logger logger = LoggerFactory.getLogger(UdpSenderService.class);
+    private final DatagramSocketWrapper datagramSocketWrapper;
+
+    public UdpSenderService(DatagramSocketWrapper datagramSocketWrapper) {
+        this.datagramSocketWrapper = datagramSocketWrapper;
+    }
+
+    public CompletableFuture<List<UdpResponse>> broadcastUpdDatagram(String content) {
+        List<String> allBroadcastAddresses = NetUtil.getAllBroadcastAddresses();
+        CompletableFuture<List<UdpResponse>> future = new CompletableFuture<>();
+        Executors.newCachedThreadPool().submit(() -> future.complete(allBroadcastAddresses.stream().map(address -> {
+            try {
+                return sendMessage(content, InetAddress.getByName(address));
+            } catch (UnknownHostException e) {
+                logger.warn("Could not find host with IP {}", address);
+                return new ArrayList<UdpResponse>();
+            }
+        }).flatMap(Collection::stream).distinct().collect(toList())));
+        return future;
+    }
+
+    public CompletableFuture<List<UdpResponse>> sendMessage(String content, String ipAddress) {
+        try {
+            CompletableFuture<List<UdpResponse>> future = new CompletableFuture<>();
+            InetAddress inetAddress = InetAddress.getByName(ipAddress);
+            Executors.newCachedThreadPool().submit(() -> future.complete(sendMessage(content, inetAddress)));
+            return future;
+        } catch (UnknownHostException e) {
+            logger.warn("Could not find host with IP {}", ipAddress);
+            return CompletableFuture.completedFuture(Collections.emptyList());
+        }
+    }
+
+    private List<UdpResponse> sendMessage(String content, InetAddress inetAddress) {
+        logger.debug("Using address {}", inetAddress);
+        byte[] buf = content.getBytes(Charset.defaultCharset());
+        DatagramPacket packet = new DatagramPacket(buf, buf.length, inetAddress, REVOGI_PORT);
+        List<UdpResponse> responses = Collections.emptyList();
+        try {
+            datagramSocketWrapper.initSocket();
+            datagramSocketWrapper.sendPacket(packet);
+            responses = getUdpResponses();
+        } catch (IOException e) {
+            logger.warn("Error sending message or reading anwser {}", e.getMessage(), e);
```

Comment:
It seems a bit unreasonable to log the stack trace when the network fails.
```suggestion
            logger.warn("Error sending message or reading anwser {}", e.getMessage());
```